plugins/services/src/js/reducers/serviceForm/FormReducers/EnvironmentVariables.js

```diff
@@ -0,0 +1,42 @@
+import {
+  ADD_ITEM,
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../../src/js/constants/TransactionTypes';
+
+module.exports = {
+  FormReducer(state = [], {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    const joinedPath = path.join('.');
+
+    if (joinedPath.search('env') !== -1) {
+      if (joinedPath === 'env') {
+        switch (type) {
+          case ADD_ITEM:
+            state.push({key: null, value: null});
+            break;
+          case REMOVE_ITEM:
+            state = state.filter((item, index) => {
+              return index !== value;
+            });
+            break;
+        }
+
+        return state;
+      }
+
+      const index = joinedPath.match(/\d+/)[0];
```

Comment:
This has turned up in multiple PR's with the same pattern I've created a issue (DCOS-15145) to tackle the problem in all reducers in one go.